creator-node/src/routes/tracks.js

```diff
@@ -306,6 +307,43 @@ router.post(
   })
 )
 
+// In total, will try for 60 seconds.
+const DEEFAULT_MAX_RETRIES = 60
+const DEFAULT_RETRY_TIMEOUT = 1000 // 1 seconds
+
+const validateTrackOwner = async ({
+  libs,
+  logger,
+  trackId,
+  userId,
+  blockNumber,
+  maxRetries = DEEFAULT_MAX_RETRIES,
+  retryTimeout = DEFAULT_RETRY_TIMEOUT
+}) => {
+  const stateDate = Date.now()
+  for (let retry = 1; retry <= maxRetries; retry++) {
+    try {
+      const discoveryTrackResponse = await libs.Track.getTracks(1, 0, [trackId])
+
+      if (
+        !Array.isArray(discoveryTrackResponse) ||
+        discoveryTrackResponse.length === 0 ||
+        !discoveryTrackResponse[0].hasOwnProperty('blocknumber')
+      ) {
+        throw new Error('Missing or malformatted track fetched from discprov.')
+      }
+      const track = discoveryTrackResponse[0]
+      if (track.blocknumber >= blockNumber) {
+        return parseInt(userId) === track.owner_id
```

Comment:
for now this is fine but we should maybe think about a DN API to return data **at** specific blocknumber e..g what if it changed at a later blocknumber? also just seems more **correct** that way